R/is_reproducible.R

```diff
@@ -0,0 +1,28 @@
+#' Reproducibility Check for `@code` Slot in `teal_data`
+#' @param teal_data `teal_data` object
+#' @examples
+#' tdata1 <- teal_data()
+#' tdata1 <- within(tdata1, {
+#'   a <- 1
+#'   b <- a^5
+#'   c <- list(x = 2)
+#' })
+#' is_reproducible(tdata1)
+#' @export
+is_reproducible <- function(teal_data) {
+  checkmate::assert_class(teal_data, "teal_data")
+  if (teal_data@valid) {
+    teal_data
+  }
+  hashes_qenv <- sapply(ls(teal_data@env), function(x) digest::digest(get(x, env = teal_data@env)))
+  eval_env <- new.env()
+  eval(parse(text = teal_data@code), envir = eval_env)
+  hashes_eval_qenv <- sapply(ls(eval_env), function(x) digest::digest(get(x, env = eval_env)))
```

Comment:
What is the benefit of using hashes over `all.equal(object@env, eval_env)`?

Reply:
Huh, didn't expect this check to be that simple

Reply:
I did use hashes because `identical()` returned FALSE for comparison of environments as it does `Exact Equality Check`. `all.equal` tests `near equality` so I guess this might do the work if it skips the environment address and just looks at environment's objects :) 

`all.equal` documentation states that 
> Do not use all.equal directly in if expressions—either use `isTRUE(all.equal(....))` or `identical` if appropriate.

but I guess `isTRUE(all.equal(` is an overkill here. 

Removed hashes with f189969a4877cd006f9ed91ff1715eeaaa3f038b